cmd/controller-manager/app/options/options.go

```diff
@@ -90,65 +88,14 @@ func NewGenericControllerManagerOptions(componentConfig componentconfig.KubeCont
 
 // NewDefaultControllerManagerComponentConfig returns default kube-controller manager configuration object.
 func NewDefaultControllerManagerComponentConfig(insecurePort int32) componentconfig.KubeControllerManagerConfiguration {
-	return componentconfig.KubeControllerManagerConfiguration{
```

Comment:
Did you make sure that all these defaults match the ones applied by the defaulter?

Reply:
Something we've found helpful with similar setups in openshift is having a sanity check unitntest that constructs a config struct, populates it from defaults (or calls this method), then DeepEqual compares it to an expected struct. Setting that unit test up against master, then making sure it passes with no changes against this PR would give me confidence we're not changing anything here. 

That also catches drift or accidental removal of defaults, which is important with config types that are in the process of being moved and are currently shared.

Reply:
@liggitt there's a `TestAddFlags` in options_test.go in this directory. it has not been touched. it does use `NewCloudControllerManagerOptions`

Reply:
> there's a TestAddFlags in options_test.go in this directory. it has not been touched. it does use NewCloudControllerManagerOptions

that's good, but that test is making sure flags override defaults, not ensuring the defaults match what is expected. if that test also compared against expected defaults before binding/parsing flags, that would cover it

Reply:
cool. will do that shortly

Reply:
Done. added a new test without parsing/binding flags. also ran it against master - http://paste.openstack.org/show/718026/ - no errors.